src/plugins/routing.js

```diff
@@ -16,13 +16,17 @@ function localePathFactory (i18nPath, routerPath) {
     locale = locale || this[i18nPath].locale
     if (!locale) return
 
+    if(STRATEGY === STRATEGIES.NO_PREFIX && locale !== this.$i18n.locale) {
+      console.warn('[<%= options.MODULE_NAME %>] localePath is unsupported when using no_prefix strategy', locale, this.$i18n.locale)
```

Comment:
It's not that `localePath` is unsupported but passing non-current locale is. So the warning should be earlier, before `locale` is overwritten so that you can check what was passed and compare to current.

Also, just printing `locale` and `this.$i18n.locale` as part of the warning, without explaining what those are is not ideal for production code. But I guess you did that just for debugging.